src/Tools/FirstRunCertGenerator/test/CertificateManagerTests.cs

```diff
@@ -247,6 +159,90 @@ public void EnsureCreateHttpsCertificate_DoesNotCreateACertificate_WhenThereIsAn
             Assert.Equal(httpsCertificate.GetCertHashString(), exportedCertificate.GetCertHashString());
         }
 
+        [Fact(Skip = "Modifies global machine state")]
+        public void EnsureCreateHttpsCertificate_ReturnsExpiredCertificateIfVersionIsIncorrect()
+        {
+            var manager = new CertificateManager();
+
+            manager.RemoveAllCertificates(CertificatePurpose.HTTPS, StoreName.My, StoreLocation.CurrentUser, TestCertificateSubject);
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                manager.RemoveAllCertificates(CertificatePurpose.HTTPS, StoreName.Root, StoreLocation.CurrentUser, TestCertificateSubject);
+            }
+
+            DateTimeOffset now = DateTimeOffset.UtcNow;
+            now = new DateTimeOffset(now.Year, now.Month, now.Day, now.Hour, now.Minute, now.Second, 0, now.Offset);
+            manager.EnsureAspNetCoreHttpsDevelopmentCertificate(now, now.AddYears(1), path: null, trust: false, subject: TestCertificateSubject);
+
+            manager.AspNetHttpsCertificateVersion = 2;
+
+            var httpsCertificateList = manager.ListCertificates(CertificatePurpose.HTTPS, StoreName.My, StoreLocation.CurrentUser, isValid: true);
+            Assert.Empty(httpsCertificateList);
+        }
+
+        [Fact(Skip = "Modifies global machine state")]
+        public void EnsureCreateHttpsCertificate_ReturnsExpiredCertificateForEmptyVersionField()
+        {
+            var manager = new CertificateManager();
+
+            manager.RemoveAllCertificates(CertificatePurpose.HTTPS, StoreName.My, StoreLocation.CurrentUser, TestCertificateSubject);
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                manager.RemoveAllCertificates(CertificatePurpose.HTTPS, StoreName.Root, StoreLocation.CurrentUser, TestCertificateSubject);
+            }
+
+            DateTimeOffset now = DateTimeOffset.UtcNow;
+            now = new DateTimeOffset(now.Year, now.Month, now.Day, now.Hour, now.Minute, now.Second, 0, now.Offset);
+            manager.AspNetHttpsCertificateVersion = 0;
+            manager.EnsureAspNetCoreHttpsDevelopmentCertificate(now, now.AddYears(1), path: null, trust: false, subject: TestCertificateSubject);
+
+            manager.AspNetHttpsCertificateVersion = 1;
+
+            var httpsCertificateList = manager.ListCertificates(CertificatePurpose.HTTPS, StoreName.My, StoreLocation.CurrentUser, isValid: true);
+            Assert.Empty(httpsCertificateList);
+        }
+
+        [Fact(Skip = "Modifies global machine state")]
+        public void EnsureCreateHttpsCertificate_ReturnsValidIfVersionIsZero()
```

Comment:
Do not skip these tests. There is no value in them if they are skipped

Reply:
My bad, I thought we shouldn't run these tests on CI. Every test except one was skipped, and the single remaining test was skipped on helix. I'll try enabling them on non-helix workloads.